web/client/components/share/SharePanel.jsx

```diff
@@ -350,6 +361,54 @@ class SharePanel extends React.Component {
                                 this.setState({...this.state, zoom});
                             }}/>
                     </FormGroup>
+                    {
+                        this.props.mapType && this.props.mapType === 'cesium' && (
+                            <React.Fragment>
+                                <FormGroup>
+                                    <ControlLabel><Message msgId="share.heading" /></ControlLabel>
+                                    <OverlayTrigger placement="top" overlay={<Tooltip id="share-zoom"><Message msgId="share.zoomToolTip"/></Tooltip>}>
+                                        <Glyphicon style={{marginLeft: 5}} glyph="info-sign" />
+                                    </OverlayTrigger>
+                                    <FormControl
+                                        type="number"
+                                        name={"heading"}
+                                        value={this.state.heading || this.props.viewerOptions.orientation.heading}
```

Comment:
```suggestion
                                        value={this.state.heading || this.props?.viewerOptions?.orientation?.heading}
```
Else on initial load of 3d map with SharePanel opened, cause app to crash